engine/src/main/java/org/hibernate/search/bridge/impl/ExtendedBridgeProvider.java

```diff
@@ -58,5 +58,13 @@ public FieldBridge provideFieldBridge(BridgeProviderContext bridgeProviderContex
 		 * not.
 		 */
 		boolean isExplicitlyMarkedAsNumeric();
+
+		/**
+		 * Returns the type of the indexed member/property; it works for arrays and collections too.
+		 *
+		 * @see #getReturnType()
+		 * @return the type of the indexed member
+		 */
+		Class<?> getExactReturnType();
```

Comment:
It's probably a good compromise until 6.0, yes. `getElementOrContainerReturnType`, then?